### PR TITLE
Add state functionality mentioned in #1011

### PR DESCRIPTION
**Pull request by justbill2020**
wasn't sure the best way to implement this i added issue #1011 and have added that functionality

/github/issues-state/:owner/:repo/:number.svg will return a result such as 
 or  depending on the current state of the issue.

### CODE REVIEW

**Comment by justbill2020** [comment]
@espadrine have you had a chance to take a look at this? I set up on a droplet the code to implement it in our dev branch https://github.com/evancohen/smart-mirror/blob/dev/README.md

the current host is https://img.myrandomwebs.com/try.html if you want to check it out there. i would love to be able to shut this server and url down though after this goes live here. 

I was also thinking about maybe making it pull the title of the issue and have that there instead of closed or open. any thoughts on that? or perhaps if `-raw` it look like it does right now... 
 or 

and if there is no `-raw` modifier it displays it with the title like this
 or

File: lib/badge.js [hunk 1]
```diff
@@ -119,8 +119,8 @@ function makeImage(data, cb) {
   if (textWidth1 % 2 === 0) { textWidth1++; }
   if (textWidth2 % 2 === 0) { textWidth2++; }
   data.widths = [
-    textWidth1 + 10 + data.logoWidth + data.logoPadding,
-    textWidth2 + 10,
+    textWidth1 + 20 + data.logoWidth + data.logoPadding,
+    textWidth2 + 30,
```
Comment (Daniel15) [inline lib/badge.js]: Why do you need to change these?

File: server.js [hunk 2]
```diff
@@ -3224,22 +3224,26 @@ cache(function(data, match, sendBadge, request) {
...
+  var format = match[8];
   var apiUrl = githubApiUrl;
   var query = {};
   var issuesApi = false;  // Are we using the issues API instead of the repo one?
   if (isPR) {
     apiUrl += '/search/issues';
     query.q = 'is:pr is:' + (isClosed? 'closed': 'open') +
       ' repo:' + user + '/' + repo;
+  } else if (isState && ghLabel != undefined) {
+    apiUrl += '/repos/' + user + '/' + repo + '/issues/' + ghLabel;
```
Comment (Daniel15) [inline server.js]: You can use a template string to build this URL, to make it look a bit cleaner:
```
apiUrl += `/repos/${user}/${repo}/issues/${ghLabel}`;
```

File: server.js [hunk 3]
```diff
@@ -3266,23 +3270,38 @@ cache(function(data, match, sendBadge, request) {
       var data = JSON.parse(buffer);
       var modifier = '';
       var issues;
-      if (isPR) {
-        issues = data.total_count;
+      if (isState && issuesApi) {
+        var rightSide = isRaw ? data.state : data.title;
+        rightSide = rightSide.length > 25 ? rightSide.substr(0,21)+'...' : rightSide
```
Comment (Daniel15) [inline server.js]: Maybe pull these out as constants (eg. `const MAX_TITLE_LENGTH = 25` or something)

I feel like this should be a separate utility function, like
```
function truncateTitle(str) {
  return str.length > MAX_TITLE_LENGTH
    ? str.substr(0, MAX_TITLE_LENGTH - 4) + '...'
    : str;
}
```

File: server.js [hunk 4]
```diff
@@ -3266,23 +3270,38 @@ cache(function(data, match, sendBadge, request) {
...
+            issues = data.length;
+            if (res.headers['link'] &&
+                res.headers['link'].indexOf('rel="last"') >= 0) {
+              modifier = '+';
+            }
+          } else {
+            issues = data.open_issues_count;
+          }
         }
+        var rightText = isRaw? '': (isClosed? ' closed': ' open');
```
Comment (Daniel15) [inline server.js]: Nit: Add one space before each `?`

**Comment by Daniel15** [comment]
There's also a lint error: https://travis-ci.org/badges/shields/builds/243122630?

Please run `npm run lint` and ensure lint is clean when submitting your changes 😃

File: lib/badge.js (hunk 1 above)
Comment (justbill2020) [inline lib/badge.js]: it was causing overlap on longer test strings from the description... if you pull it and run it with old values you'll see what i mean.

File: server.js (hunk 2 above)
Comment (justbill2020) [inline server.js]: for sure!

File: server.js (hunk 3 above)
Comment (justbill2020) [inline server.js]: I like this too...

File: server.js (hunk 4 above)
Comment (justbill2020) [inline server.js]: good call... i was typing fast to get this all done so we could host and add functionality

**Comment by justbill2020** [comment]
I have been super swamped with my full time gig and haven't had a chance to make revisions requested I should have time at some point next week

**Comment by paulmelnikow** [comment]
If anyone would like to adopt this PR, please feel free to do so: grab the branch, add more commits addressing the comments, and open a new PR referencing this one. Will close in one month if no one has done so.

**Comment by Coeur** [comment]
Can we have three states instead of two?
* Open
* Closed
* Merged

This way we are compatible with pull requests as well. Example of pull request disguised as an issue:
https://github.com/hashicorp/terraform/issues/708

**Comment by justbill2020** [comment]
I'm so sorry about allowing this to go stale. I started a new job and between that and the fam the free time has been little to none. I do have a version of this with those changes on a pc at home that I haven't committed or pushed yet. Im currently traveling and should be able to push it when I get home.

**Comment by paulmelnikow** [comment]
No problem! We're all volunteering our time, and life happens.

Shields needs something similar to this for our CI (see #979) and I could work on them together. If you don't mind, I'll pick this up.

**Comment by justbill2020** [comment]
no problem... pick it up... let me know if you need any questions or what not answered

**Comment by paulmelnikow** [comment]
Will do. Thanks!

**Pull request closed without merging**